nbextensions/bulkImportCell/tabs/configure.js

```diff
@@ -197,35 +202,87 @@ define([
              * 2 - eval the array of file inputs and outputs.
              * If both are up to snuff, we're good.
              */
+            const otherParamIds = model.getItem(['app', 'otherParamIds', fileType]),
+                paramValues = model.getItem(['params', fileType, PARAM_TYPE]),
+                filePathIds = model.getItem(['app', 'fileParamIds', fileType]),
+                filePathValues = model.getItem(['params', fileType, FILE_PATH_TYPE]);
+
+            const filePathValidations = filePathValues.map((filePathRow) => {
+                return spec.validateParams(filePathIds, filePathRow);
+            });
+            return Promise.all([
+                spec.validateParams(otherParamIds, paramValues),
+                ...filePathValidations,
+            ]).then((results) => {
+                let isValid = true;
+                results.forEach((result) => {
+                    Object.values(result).forEach((param) => {
+                        if (!param.isValid) {
+                            isValid = false;
+                        }
+                    });
+                });
+                return isValid ? 'complete' : 'incomplete';
+            });
         }
 
         /**
+         * Updates the stored parameter value, for the given fileType and parameter type based on
+         * the bus message passed by the input widget.
          *
-         * @param {string} paramKey - a string with the parameter key
+         * This is followed by evaluating the state of the set of inputs for this file type and
+         * passing a message to the controller if there's a change.
+         * @param {string} fileType - the current filetype this tab is investigating
          * @param {string} paramType - should be either 'param' or 'filePath'
-         * @param {object} message -
+         * @param {object} message - the bus message returned from the widget with a changed
+         *  parameter value
          */
-        function updateModelParameterValue(paramKey, paramType, message) {
-            model.setItem(['params', paramKey, paramType, message.parameter], message.newValue);
-            // TODO - update the app config state based on changes, and send a message up the cellBus
-            // with that state.
-            const appState = evaluateAppConfig(message.isError);
+        function updateModelParameterValue(fileType, paramType, message) {
+            if (paramType === FILE_PATH_TYPE) {
+                model.getItem(['params', fileType, FILE_PATH_TYPE])[message.rowIndex][
+                    message.parameter
+                ] = message.newValue;
+            } else {
+                model.setItem(['params', fileType, paramType, message.parameter], message.newValue);
+            }
+
+            // evaluate the parameter state for the current file type
+            return Promise.try(() => {
+                if (message.isError) {
+                    return 'error';
```

Comment:
I think the difference between "error" and "incomplete" is just semantic. An "error" means an invalid input, like trying to name a new object with illegal characters. "Incomplete" means a required field is missing.